### PR TITLE
[GraphQL] Rework Event type

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Accessing the `bcs` field of an `Event` is rather confusing, as it is not the whole event BCS encoded, but just the contents of it, more specifically, the BCS bytes of a `MoveValue`.

This PR removes the #flatten in `Event`, adds a `contents` field to the type, and adds the `bcs` field to the `Event` type which encodes the whole `Event`.

## Test plan 

Existing tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [x] GraphQL: The `Event` type has a new field called `contents`, which is the event's content value as a `MoveValue`. This replaces the previous scheme that flattened the `MoveValue` type in the `Event` type. A `bcs` field was also added, which represents the Base64 encoded BCS serialized event.
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]

**Review by amnn — APPROVED** [review]
Nice, thank you @stefan-mysten ! Could you also give a heads up to the SDK working group about this change and when it's expected to be released?

**Comment by stefan-mysten** [comment]
> Nice, thank you @stefan-mysten ! Could you also give a heads up to the SDK working group about this change and when it's expected to be released?

Definitely, thanks!

**Review by hayes-mysten — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
         run: pnpm dlx concurrently --kill-others --success command-1 "$E2E_RUN_LOCAL_NET_CMD" 'pnpm --filter @mysten/sui test:e2e'
 
       - name: Run RPC/GrpahQL comaptability e2e tests
-        if: ${{ needs.diff.outputs.isGraphQlTransport == 'true' || needs.diff.outputs.isRust == 'true'}}
+        if: ${{ (needs.diff.outputs.isGraphQlTransport == 'true' || needs.diff.outputs.isRust == 'true') && github.ref == 'refs/heads/main' }}
```
Comment (hayes-mysten) [inline .github/workflows/e2e.yml:74]: this will also disable this on all PRs right?

Comment (hayes-mysten) [reply .github/workflows/e2e.yml:74]: we probably still want this for PRs, but just against localnet

Comment (stefan-mysten) [reply .github/workflows/e2e.yml]: I pushed another one that checks specifically for devnet and testnet branches. I think that should work, what do you think? @hayes-mysten

Comment (hayes-mysten) [reply .github/workflows/e2e.yml]: I am not very familiar with the github workflow syntax, but seems reasonable to me!

**Pull request merged**